### PR TITLE
Stop recommending insecure IMDSv2 configuration

### LINKED ISSUE #2665: IMDSv2 Doc Suggests Insecure Configuration
**Describe the bug**
The [IMDSv2 section of the Installation Guide](https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.4/deploy/installation/#using-metadata-server-version-2-imdsv2) suggests increasing the endpoint hop limit to 2, despite this being directly against the recommendation of [EKS best practices](https://aws.github.io/aws-eks-best-practices/security/docs/iam/#restrict-access-to-the-instance-profile-assigned-to-the-worker-node). This Best Practice is relevant even for non-EKS clusters running in AWS.

This is also without any suggestion that this could seriously impact a cluster's security posture.

**Steps to reproduce**
- read the 2 linked documents

**Expected outcome**
A removal of the recommendation, or at least a warning about the security implications, and a guide on a minimally-scoped NetworkPolicy to allow _only_ the AWS LB Controller minimal access.

**Environment**

* AWS Load Balancer controller version N/A
* Kubernetes version N/A
* Using EKS (yes/no), if so version? N/A

**Additional Context**:

### PR DESCRIPTION
**Pull request by johngmyers**
### Issue

Fixes #2665 

### Description

Update installation documentation to replace the advice to configure IMDSv2 hops insecurely.

### Checklist
- [ ] Added tests that cover your change (if possible)
- [x] Added/modified documentation as required (such as the `README.md`, or the `docs` directory)
- [ ] Manually tested
- [x] Made sure the title of the PR is a good description that can go into the release notes

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

**Review by olemarkus — APPROVED** [review]

File: docs/deploy/installation.md [hunk 1]
```diff
@@ -20,15 +20,14 @@ The LBC is supported by AWS. Some clusters may using legacy "in-tree" functional
...
+### Determining VPC and Region
 
-You can set the IMDSv2 hop limit as follows:
-```
-aws ec2 modify-instance-metadata-options --http-put-response-hop-limit 2 --region <region> --instance-id <instance-id>
-```
+The controller must be able to determine the AWS Region and VPC ID. There are several options:
 
-Instead of depending on IMDSv2, you alternatively may specify the AWS region and the VPC via the controller flags `--aws-region` and `--aws-vpc-id`.
+1. You may specify either or both using the controller flags `--aws-region` and/or `--aws-vpc-id`.
```
Comment (kishorj) [inline docs/deploy/installation.md:27]: With these instructions in place, the current default installation will not end up with a functional controller (assuming the hop limit is 1). I'm more comfortable with these instructions once we are able to determine VPC from the node resource without the node name limitation.

Comment (johngmyers) [reply docs/deploy/installation.md:27]: Perhaps we should make the default installation specify host networking?

Comment (M00nF1sh) [reply docs/deploy/installation.md:27]: @johngmyers 
I think host networking provides unnecessary permissions to the controller, which is not secure.

Chatted with @kishorj , the best solution to moving forward is to auto-detect the region & vpc-ID to remove IMDS dependency for default installation.

Comment (johngmyers) [reply docs/deploy/installation.md:27]: @M00nF1sh Setting the IMDSv2 hop limit to 2 is even less secure than giving host networking to LBC, by a lot. That provides unnecessary permissions to every pod on the node, allowing them to obtain credentials to the node's instance role.

**Pull request closed without merging**